### PR TITLE
extended existing HLT tau-jet correlation module [13_0_X]

### PR DESCRIPTION
**Pull request by brallmond**
#### PR description:
To support a future VBF Single Tau HLT in the scenario that VBF Parking is accepted (see JIRA https://its.cern.ch/jira/browse/CMSLITDPG-1099 and https://its.cern.ch/jira/browse/CMSHLT-2702), a small change to an existing HLT module is necessary. The change extends the module's usage from only handling 2 Jet + 2 Tau events to also handling 2 Jet + 1 Tau events. This change depends on no other PRs and affects code used by one existing HLT path, and would allow the implementation of a future HLT path.  

#### PR validation:
In the course of estimating rate and gain for the proposed VBF Single Tau path, the same metrics were also computed for the VBF DiTau path with the module extension in place and no differences were found in past and current behavior of the path. 

#### Backport 
master PR is https://github.com/cms-sw/cmssw/pull/40993
It's necessary to Backport this change as current testing and HLT development is taking place in CMSSW_13_0_0

### CODE REVIEW

**Comment by Martin-Grunewald** [comment]
Please make a PR to the master CMSSW branch as well so that this backport can be integrated with the master being up to date as well.

File: RecoTauTag/HLTProducers/src/HLTPFDiJetCorrCheckerWithDiTau.cc [hunk 1]
```diff
@@ -84,17 +87,21 @@ void HLTPFDiJetCorrCheckerWithDiTau::produce(edm::StreamID iSId, edm::Event& iEv
...
-            if (reco::deltaR2(taus[iTau2]->p4(), myPFJet1.p4()) < dRmin2_)
-              continue;
-            if (reco::deltaR2(taus[iTau2]->p4(), myPFJet2.p4()) < dRmin2_)
-              continue;
-
-            correctComb = true;
-            break;
+          if (taus.size() == 1) { correctComb = true; }
+
+          if (taus.size() > 1) {
```
Comment (Martin-Grunewald) [inline RecoTauTag/HLTProducers/src/HLTPFDiJetCorrCheckerWithDiTau.cc:92]: just use  ```else``` since you checked size()>0 already above....

**Comment by brallmond** [comment]
> Please make a PR to the master CMSSW branch as well so that this backport can be integrated with the master being up to date as well.

Okay, when I tried to integrate with master originally, there were 78+ commits included from other PRs. Is this intended?

**Comment by Martin-Grunewald** [comment]
You'd need to rebase to 13_1_X or start from a recent 13_1_X developer area to make the PR.

**Comment by brallmond** [comment]
> You'd need to rebase to 13_1_X or start from a recent 13_1_X developer area to make the PR.

Okay, I've made a new PR (https://github.com/cms-sw/cmssw/pull/40993) to 13_1_X with the same change, and edited the description of the PR in this thread to show it is a backport to 13_0_0. Please let me know if this is acceptable

**Comment by Martin-Grunewald** [comment]
OK, looks godd - but please apply the 131x code format patches there and also here.

**Comment by perrotta** [comment]
backport of #40993

**Comment by missirol** [comment]
please test

**Comment by missirol** [comment]
+hlt

 - verbatim backport of #40993 
 - see https://github.com/cms-sw/cmssw/pull/40993#issuecomment-1462657849

**Pull request merged**